### ISSUE TITLE
2XSC v8.1.3 - Improve missing file error handling

### ISSUE THREAD

**Issue by tpperlman** [2sic/2sxc#656]
When a user attempts to edit a data item when a linked file is missing, they receive an unfriendly error message that doesn't help diagnose the issue.

![missing file](https://cloud.githubusercontent.com/assets/1871915/13082746/f4faed9e-d49f-11e5-9f94-08da76347cf9.JPG)

It is my opinion that if SXC is designed to make content editing dummy-proof, it should provide better exceptions than "500 Errror". When users receive this, this is what they report to us as website admins. We can't count on users to document everything they did step-by-step to get there. Short of that, we as website admins will have to try to diagnose the issues before we can try and resolve or even report them here. I think it is vital that any content editor features should provide as much error notification as possible so that it truly can become the best content management feature in DNN.

**Comment by iJungleboy** [comment]
To add a few specs, I believe the correct implementation should

1. not show a dialog - to not interrupt the user, because if there would be multiple of these, it would feel very annoying
2. somehow flag the file/hyperlink field as faulty and show an error somewhere there
3. still allow saving - because sometimes the current editing user won't know how to fix this, so it shouldn't prevent the user from making other changes

Note that this will probably benefit from the #595

**Comment by iJungleboy** [comment]
I wanted to fix this issue but I can't reproduce it. When I have a file-field and delete the file + re-open the dialog, I don't get a JS error. In which case does this happen? pls give me steps to reproduce.

**Comment by iJungleboy** [comment]
In the next release you'll also see more details in the JS error message. May this is already done then, will close. If it's still important, please provide more datails and re-open.
